frontend/src/lib/user.ts

```diff
@@ -115,7 +115,25 @@ function clear(): void {
 
 export async function hash(password: string): Promise<string> {
   const msgUint8 = new TextEncoder().encode(password) // encode as (utf-8) Uint8Array
-  const hashBuffer = await crypto.subtle.digest('SHA-1', msgUint8) // hash the message
+  let hashBuffer: ArrayBuffer;
+  // const c = crypto ? crypto : await import('node:crypto');
+  if (crypto && crypto.subtle) {
+    hashBuffer = await crypto.subtle.digest('SHA-1', msgUint8) // hash the message
+  } else {
+    // Running server-side
+    try {
+      const crypto = await import('node:crypto');
+      if (crypto.subtle)
+        hashBuffer = await crypto.subtle.digest('SHA-1', msgUint8) // hash the message
+      else {
+        console.log('crypto is', crypto);
+        throw new Error('crypto found but crypto.subtle not there');
+      }
+    } catch (err) {
+      console.log('ERROR: crypto module not available -- are we running on an old version of Node? Error was', err);
+      hashBuffer = new ArrayBuffer(0); // Should cause the hash comparison to fail
```

Comment:
Shouldn't we just throw the error again?
If we ever land in here we want to know about it right?

Reply:
As mentioned above, I'm going to make a change that will end up throwing the exception. I put this part in, swallowing the exception, so that I could actually get `docker build` to succeed, otherwise I was being blocked on my currrent work.